Code/Engine/ECS/Components/ComponentTypes.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include "LineRendererComponent.h"
+#include "ControllerComponent.h"
+#include "RenderableObjectComponent.h"
+#include "RenderableResourceComponent.h"
+#include "TransformComponent.h"
+
+/*
+#define COMPONENT_TYPES \
+sad::ecs::LineRendererComponent,\
+sad::ecs::RenderableObjectComponent, \
+*/
```

Comment:
Is this required?

If we wanted to document components required at startup vs. those generated at runtime we should probably make documentation for it. Or at least make this distinction clearer in a code comment.